tools/cellprofiler/common.xml

```diff
@@ -0,0 +1,395 @@
+             "HasImagePlaneDetails:False",
+             "\n"]
+
+  f.writelines(headers)
+
+  write_images()
+  write_metadata()
+  write_nameandtypes()
+  write_groups()
+
+  f.close()
+
+    </configfile>
+  </configfiles>
+  <command><![CDATA[
+    	  python $script_file $inputs
+  ]]></command>
+  <inputs>
+    
+<!-- Images module-->
+    <section name="images" title="Images" expanded="false">
+      <param name="filter_images" type="select" label="Filer images?">
+        <option value="Images only">Images only</option>
+        <option value="No filtering">No filtering</option>
+      </param>
+    </section>
+
+<!-- Metadata module-->
+    <section name="metadata" title="Metadata" expanded="false">
+      <conditional name="con_metadata_extraction">
+        <param name="extract" type="select" display="radio" label="Extract metadata?">
+           <option value="No">No</option>
+           <option value="Yes">Yes</option>
+        </param>
+        <when value="Yes">
+          <repeat name="extraction_method" title="metadata">
+              <param name="metadata_extraction_method" type="select" label="Metadata extraction method">
+                <option value="Extract from file/folder names">Extract from file/folder names</option>
+                <option value="Import from file">Import from file</option>
+                <option value="Extract from image file headers">Extract from image file headers</option>
+              </param>
+              <param name="metadata_source" type="select" label="Metadata source">
+                <option value="File name">File name</option>
+                <option value="Folder name">Folder name</option>
+              </param>   
+              <param name="file_name_regex" type="text" value="^(?P&lt;Plate>.*)_(?P&lt;Well>[A-P][0-9]{2})_s(?P&lt;Site>[0-9])_w(?P&lt;ChannelNumber>[0-9])" label="Regular expression to extract from file name">    
+                  <sanitizer sanitize="false"/>
+              </param>
+              <param name="folder_name_regex" type="text" value="(?P&lt;Date>[0-9]{4}_[0-9]{2}_[0-9]{2})$" label="Regular expression to extract from folder name"> 
+                  <sanitizer sanitize="false"/>
+              </param>
+
+              <param name="extract_metadata_from" type="select" label="Extract metadata from">
+                <option value="All images">All images</option>
+                <option value="Images matching a rule">Images matching a rule</option>
+              </param>  
+          </repeat>  
+          <param name="metadata_type" type="select" label="Metadata data type">      
+                  <option value="Text">Text</option>
+          </param>
+        </when>
+      </conditional>
+    </section>
+
+<!-- NamesAndTypes-->    
+    <section name="nameandtypes" title="NamesAndTypes" expanded="false">
+        <conditional name="pixel_space">
+            <param name="process_3d" type="select" display="radio" label="Process 3D">

```

Comment:
I would remove the radio here ... its also in https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices.html

Reply:
i just thought may be better to align with Cellprofiler UI as much as possible..
but I can still remove the radio, let me know what you would suggest. thanks